File: Assets/zfoolua/jprotobuf/JProtobufHelloRequest.lua

local JProtobufHelloRequest = {}

function JProtobufHelloRequest:new()
    local obj = {
        message = "" -- string
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function JProtobufHelloRequest:protocolId()
    return 1500
end

function JProtobufHelloRequest:write(buffer, packet)
    if packet == nil then
        buffer:writeInt(0)
        return
    end
    buffer:writeInt(-1)
    buffer:writeString(packet.message)
end

function JProtobufHelloRequest:read(buffer)
    local length = buffer:readInt()
    if length == 0 then
        return nil
    end
    local beforeReadIndex = buffer:getReadOffset()
    local packet = JProtobufHelloRequest:new()
    local result0 = buffer:readString()
    packet.message = result0
    if length > 0 then
        buffer:setReadOffset(beforeReadIndex + length)
    end
    return packet
end

return JProtobufHelloRequest

File: Assets/zfoocs/Jprotobuf/JProtobufHelloRequest.cs
using System;
using System.Collections.Generic;

namespace zfoocs
{
    
    public class JProtobufHelloRequest
    {
        public string message;

        public static JProtobufHelloRequest ValueOf(string message)
        {
            var packet = new JProtobufHelloRequest();
            packet.message = message;
            return packet;
        }
    }


    public class JProtobufHelloRequestRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 1500;
        }

        public void Write(ByteBuffer buffer, object packet)
        {
            if (packet == null)
            {
                buffer.WriteInt(0);
                return;
            }
            JProtobufHelloRequest message = (JProtobufHelloRequest) packet;
            buffer.WriteInt(-1);
            buffer.WriteString(message.message);
        }

        public object Read(ByteBuffer buffer)
        {
            int length = buffer.ReadInt();
            if (length == 0)
            {
                return null;
            }
            int beforeReadIndex = buffer.ReadOffset();
            JProtobufHelloRequest packet = new JProtobufHelloRequest();
            string result0 = buffer.ReadString();
            packet.message = result0;
            if (length > 0) {
                buffer.SetReadOffset(beforeReadIndex + length);
            }
            return packet;
        }
    }
}
